### PR TITLE
Re-enable ORCA tests disabled by PR #11336

### PR DESCRIPTION
**Pull request by hardikar**
https://github.com/greenplum-db/gpdb/pull/11336 disable almost all of
ORCA's "unit" tests a.k.a ctests because they involved a large amount of
work to modify to work with the new implementation of handling partitioned
tables in ORCA.
This PR includes the changes needed to update the metadata & query objects
as well as commit the resultant plan changes. The metadata (partition table
entries and child partition information) for many MDPs had to be parsed from
the DLL in the comment section of the MDP, and when that was missing,
it was guessed or manually constructed.

This PR also includes some minor fixes to get the tests compiling and 2 reverts.

### CODE REVIEW

**Review by chrishajas — APPROVED** [review]
Overall LGTM. I took a look at the non-mdp files and a couple of the mdps. Would it be worth including the script you used to convert these somewhere (even if the logic might not be useful in the future, but the architecture/parsing may)?

File: src/backend/gporca/libgpopt/src/base/CDefaultComparator.cpp [hunk 1]
```diff
@@ -86,6 +86,56 @@ CDefaultComparator::FEvalComparison(CMemoryPool *mp, const IDatum *datum1,
...
+	{
+		// For now, specifically target date and timestamp columns, since they
+		// are mappable to a double value that represents the number of microseconds
+		// since Jan 1, 2000 and therefore those can be compared precisely, just like
+		// integer types
+		if (!(CMDIdGPDB::m_mdid_date.Equals(mdid1) ||
+			  CMDIdGPDB::m_mdid_time.Equals(mdid1) ||
+			  CMDIdGPDB::m_mdid_timestamp.Equals(mdid1)))
+		{
+			result = false;
```
Comment (chrishajas) [inline src/backend/gporca/libgpopt/src/base/CDefaultComparator.cpp:127]: With this, are we allowing the internal evaluator to process dates/times/timestamps for both gpdb queries and orca tests?

Comment (hardikar) [reply src/backend/gporca/libgpopt/src/base/CDefaultComparator.cpp:127]: We are. I'm going to pull it out as a separate story.

**Comment by d** [comment]
Sweet, I assume once this is integrated I should rebase #11403 on top of it?

**Comment by d** [comment]
> had to be parsed from
> the DLL in the comment section of the MDP, and when that was missing,
> it was guessed or manually constructed.

I think you meant "DXL", or "XML", instead of "DLL"?

**Pull request merged**